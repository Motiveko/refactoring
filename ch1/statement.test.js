import fs from 'fs';
import {htmlStatement, statement} from './statment.js';

const invoicies = JSON.parse(fs.readFileSync('ch1/invoicies.json'));
const plays = JSON.parse(fs.readFileSync('ch1/plays.json'));


const [invoice] = invoicies;
test('test statement', () => {
  expect(statement(invoice, plays)).toMatchSnapshot();
})

test('test htmlStatement', () => {
  expect(htmlStatement(invoice, plays)).toMatchSnapshot();
})