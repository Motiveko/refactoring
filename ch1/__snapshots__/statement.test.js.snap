// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test htmlStatement 1`] = `
"<h1>청구 내역 (고객명: BigCo)</h1>
<table>
<tr><th>연극</th><th>좌석 수</th><th>금액</th></tr><tr><td>Hamlet</td>(55석)<td></td><td>$650.00</td></tr><tr><td>As You Like it</td>(35석)<td></td><td>$580.00</td></tr><tr><td>Othello</td>(40석)<td></td><td>$500.00</td></tr></table>
<p>총액: <em>$1,730.00</em></p>
<p>적립 포인트: <em>47점</em></p>
"
`;

exports[`test statement 1`] = `
"청구 내역 (고객명: BigCo)
Hamlet: $650.00 (55석)
As You Like it: $580.00 (35석)
Othello: $500.00 (40석)
총액: $1,730.00
적립 포인트: 47점
"
`;
